### DISCUSSION TITLE
How to create a histogram profile in a root file using uproot?

### DISCUSSION THREAD

**Discussion by rmelchi** [scikit-hep/uproot5#932] [Q&A]
I am new to uproot, but I have noticed that it is able to recognize histogram types in files created by Root. I was wondering if it is possible to create an histogram profile with uproot and store it in a root file.

Thanks

**Comment by jpivarski** [comment]
Ideally, you'd want to make a profile-style histogram with [hist](https://hist.readthedocs.io/) (maybe the [WeightedMean Storage type](https://hist.readthedocs.io/en/latest/user-guide/storages.html#weightedmean)) and assign that into an [uproot.WritableDirectory](https://uproot.readthedocs.io/en/latest/uproot.writing.writable.WritableDirectory.html). Unfortunately, it doesn't look like Uproot recognizes that hist type directly:

https://github.com/scikit-hep/uproot5/blob/130c55bdfa64426fdc6b2078c2a379dc449c1584/src/uproot/writing/identify.py#L248-L251

If Uproot had recognized this histogram type, it would use `uproot.writing.identify.to_TProfile` to convert it into ROOT format. That function is available, though it has a lot of arguments (for all of the data that _would_ come from the hist object):

https://github.com/scikit-hep/uproot5/blob/130c55bdfa64426fdc6b2078c2a379dc449c1584/src/uproot/writing/identify.py#L1552-L1593

The problem to be solved is knowing how to convert from the profile-like histogram you have into the arguments for this function. Once that's solved, the output of this function can be assigned directly into the directory:

```python
with uproot.recreate("some_file.root") as file:
    file["some_profile"] = uproot.writing.identify.to_TProfile(...)
```

Conversion in the other direction exists: if Uproot extracts a `TProfile` object from a ROOT file and you call `to_boost` on it (hist is a subclass of [boost-histogram](https://boost-histogram.readthedocs.io/)), the data members like `fBinEntries` and `fBinSumw2` get converted into the hist/boost-histogram internal data.

https://github.com/scikit-hep/uproot5/blob/130c55bdfa64426fdc6b2078c2a379dc449c1584/src/uproot/behaviors/TProfile.py#L296-L332

You'd want to reverse that transformation.

If you do solve this inverse transformation, we'd be in a good position to add it to uproot/writing/identify.py, so that there won't be a `NotImplementedError` anymore...
